js/server/process-lm.js

```diff
@@ -79,13 +76,13 @@ function processUserEvents (users, events) {
         user.tickets
       );
       const { claimed, forfeited } = calculateClaimReward(burnedTickets);
-      user.claimed += claimed;
+      user.claimableRewardsOnWithdrawnAssets += claimed;
```

Comment:
This is used as `claimed` in the UI. And, to my understanding, this increments on liquidity withdraws AND claim events from DEX.

Reply:
Sorry, I didn't specify which keys we currently use in the DEX:

```
claimed
claimableReward
dispensed
maturityDate
nextRewardProjectedAPYOnTickets
totalRewardAtMaturity
```

Reply:
To document what we discussed on call: Properties were renamed for clarity on the backend. i.e. in this case, the user did not actually claim anything, and we wouldn't want to reject an actual claim event on the basis of that amount already being included in `claimed`. 

Before deployment, the API layer will reformat the response data to accommodate existing usage on the front-end. I am keeping an eye on [RewardsPage.vue](https://github.com/Sifchain/sifnode/blob/develop/ui/app/src/views/RewardsPage.vue#L39) to ensure seamless integration on deployment to mainnet.

Reply:
Cool

Reply:
Yea I think without TypeScript it helps to make these names as clear as possible (on backend).

Reply:
Also making note to keep an eye on [RewardsContainer.vue](https://github.com/Sifchain/sifnode/pull/1323/files)

Reply:
@59023g Yeah I actually started adding TypeScript to part of it, as I think it would make the project much more maintainable and reduce bugs without having to wait for execution. But then I checked @ajoslin and @musnit 's GH profiles & didn't see any TypeScript (accurate analysis?) so I `git reset` to keep it contributor friendly

Reply:
i think typescript would make sense longer term but given we have a short term need here and the long term relevance of this code is not yet clear and may even be discarded in a few weeks, not sure it's worth the investment atm